### ISSUE TITLE
MedicationRequest medication entry not  used

### ISSUE THREAD

**Issue by sorlioglu** [cqframework/cql-exec-fhir#43]
When we wrap a MedicationRequest with ```cqlfhir.FHIRWrapper.FHIRv401()```

```
 {
                    "resourceType": "MedicationRequest",
                    "id": "04223fb9-ce82-4395-9db9-eac183bdc746",
                    "meta": {
                        "versionId": "1696647701000",
                        "lastUpdated": "2023-10-06T23:01:41.000-04:00"
                    },
                    "text": { // text rmoved for brevity purposes
                    },
                    "status": "stopped",
                    "intent": "order",
                    "priority": "routine",
                    "medicationReference": {
                        "reference": "Medication/ef21e81f-58b4-4229-a7bf-3dea06200b45",
                        "type": "Medication",
                        "display": "Metformin"
                    },
                    "medication": {
                       "coding": [
                           {
                               "system": "http://www.nlm.nih.gov/research/umls/rxnorm",
                               "code": "1313112",
                           }
                       ]
                    },
                    "subject": {
                        "reference": "Patient/b2c4558a-9e06-492f-bfe7-33eb67804dfa",
                        "type": "Patient",
                        "display": "Thomas Smith (OpenMRS ID: 10017E)"
                    },
                    "encounter": {
                        "reference": "Encounter/223876cb-e1d8-4266-a0b9-94c7285345dd",
                        "type": "Encounter"
                    },
                    "authoredOn": "2023-10-06T23:01:41-04:00",
                    "requester": {
                        "reference": "Practitioner/f9badd80-ab76-11e2-9e96-0800200c9a66",
                        "type": "Practitioner",
                        "identifier": {
                            "value": "UNKNOWN"
                        },
                        "display": "Super User (Identifier: UNKNOWN)"
                    },
                   // Other entries removed for brevity purposes

```

Only `medicationReference` get's wrapped and `medication` (with coding information) is ignored. Like so...

```
{
    "status": {
      "value": "stopped"
    },
    "intent": {
      "value": "order"
    },
    "priority": {
      "value": "routine"
    },
    "medication": {
      "reference": { // No coding information, just reference
        "value": "Medication/ef21e81f-58b4-4229-a7bf-3dea06200b45"
      },
      "type": {
        "value": "Medication"
      },
      "display": {
        "value": "Phenytoin"
      }
    },
    "subject": {
      "reference": {
        "value": "Patient/b2c4558a-9e06-492f-bfe7-33eb67804dfa"
      },
      "type": {
        "value": "Patient"
      },
      "display": {
        "value": "Thomas Smith"
      }
    },
    "encounter": {
      "reference": {
        "value": "Encounter/223876cb-e1d8-4266-a0b9-94c7285345dd"
      },
      "type": {
        "value": "Encounter"
      }
    },
    "authoredOn": {
      "value": "2023-10-06T23:01:41-04:00"
    },
    "requester": {
      "reference": {
        "value": "Practitioner/f9badd80-ab76-11e2-9e96-0800200c9a66"
      },
      "type": {
        "value": "Practitioner"
      },
      "identifier": {
        "value": {
          "value": "UNKNOWN"
        }
      },
      "display": {
        "value": "Super User (Identifier: UNKNOWN)"
      }
    },
    
    "text": {
    // Removed for brevity purposes
     
    },
    "id": {
      "value": "04223fb9-ce82-4395-9db9-eac183bdc746"
    },
    "meta": {
      "versionId": {
        "value": "1696647701000"
      },
      "lastUpdated": {
        "value": "2023-10-06T23:01:41.000-04:00"
      }
    }
  }
```

TLDR: Only ```medicationReference``` is wrapped with ```cqlfhir.FHIRWrapper.FHIRv401()``` and not ```medication``` on a MedicationRequest

**Comment by cmoesel** [comment]
Hi @sorliog. If you look at the definition of [FHIR R4 MedicationRequest](https://hl7.org/fhir/R4/medicationrequest.html), `medication[x]` is a choice:

![CleanShot 2023-11-20 at 09 23 34@2x](https://github.com/cqframework/cql-exec-fhir/assets/2278253/a786f648-eb72-4215-ab34-16e8f1caf435)

This means you must choose between sending the medication as a reference (via `medicationReference`) or as a code (via `medicationCodeableConcept`). You cannot send both at the same time unless you are using FHIR R5, but no one has published a CQL model for FHIR R5 yet -- so it's not possible at this time.

**Comment by sorlioglu** [comment]
Thank you, @cmoesel I do realize that. However, even when I try to send the medication without the medicationReference in the resource, the problem still persists.

**Comment by cmoesel** [comment]
Are you sending it as `"medicationCodeableConcept"`? In your example above, you just had `"medication"`.

**Comment by sorlioglu** [comment]
Just ```medication```. As demonstrated [here](https://www.hl7.org/fhir/medicationrequest0324.json.html)

**Comment by cmoesel** [comment]
That example is from FHIR R5. You are using FHIR R4, so you need to look at the corresponding example in FHIR R4 [here](https://hl7.org/fhir/R4/medicationrequest0324.json.html). Note that it uses `medicationCodeableConcept`.

**Comment by sorlioglu** [comment]
@cmoesel Thank you. That was in fact the solution.
